pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -438,6 +446,121 @@ func (o *InstallOptions) Run(ctx context.Context) error {
 	return nil
 }
 
+func (o *InstallOptions) waitForOtherInstallerRevisionsToSettle(ctx context.Context) error {
+	currRevision, err := strconv.ParseInt(o.Revision, 10, 32)
+	if err != nil {
+		return fmt.Errorf("bad local revision %v: %w", o.Revision, err)
+	}
+
+	// at this point we have a list of all the installer pods present on this node we need to check
+	// 1. is there a revision later than us present right now.  If yes, exit.
+	// 2. is there an installer pod present that has not finished running, wait until it is finished.
+	err = wait.PollImmediateWithContext(ctx, 10*time.Second, 60*time.Second, func(ctx context.Context) (done bool, err error) {
+		installerPods, err := o.getInstallerPodsOnThisNode(ctx)
+		if err != nil {
+			return false, nil
+		}
+		if len(installerPods) == 0 {
+			return false, fmt.Errorf("no installer pods found")
+		}
+		latestRevision, err := getRevisionOfPod(installerPods[len(installerPods)-1])
+		if err != nil {
+			return false, err
+		}
+		if latestRevision > int(currRevision) {
+			return false, fmt.Errorf("more recent revision present on node: thisRevision=%v, moreRecentRevision=%v", currRevision, latestRevision)
```

Comment:
if a newer revision is placed on the node while an older revision is in this state, then the older revision quits here.